app/connectors/MinimalDetailsConnector.scala

```
@@ -65,7 +64,10 @@ class MinimalDetailsConnectorImpl @Inject() (appConfig: FrontendAppConfig, http:
         case e @ WithStatusCode(FORBIDDEN) if e.message.contains(Constants.delimitedPSA) =>
           Left(DelimitedAdmin)
       }
-      .tapError(t => Future.successful(logger.error(s"Failed to fetch minimal details with message ${t.getMessage}")))
+      .recoverWith { t =>
+        logger.error(s"Failed to fetch minimal details with message ${t.getMessage}", t)
+        Future.failed(t)
```

Previously this was Future.successful now it is Future.failed. Should we keep it successful?
(same for all other places below)

---

Hi @kberec , I think we can’t keep Future.successful here because it converts failures into successes and hides errors.

---

After reviewing the contents of the tapError util function, I can confirm it is discarding the Future.successful result returned here and instead returning Future.failed. The purpose of the function passed into tapError is just to log, the return value is not used.

the Util uses both flatMap and recoverWith, not sure why it uses both...